### PR TITLE
Implemented Context methods and transaction Isolation levels

### PR DESCRIPTION
**Pull request by denisenkom**
Tested on go 1.8 and 1.6

### CODE REVIEW

File: tran.go [hunk 1]
```diff
@@ -16,6 +16,15 @@ const (
 	tmSaveXact      = 9
 )
 
+const (
+	isolationUseCurrent = 0
```
Comment (kardianos) [inline tran.go]: perhaps assign it a type of uint8 or `type isoLevel uint8` and use throughout, one less cast in one place.

File: mssql.go [hunk 2]
```diff
@@ -88,11 +89,15 @@ func (c *MssqlConn) Rollback() error {
 }
 
 func (c *MssqlConn) Begin() (driver.Tx, error) {
+	return c.begin(isolationUseCurrent)
+}
+
+func (c *MssqlConn) begin(tdsIsolation uint8) (driver.Tx, error) {
```
Comment (kardianos) [inline mssql.go]: maybe `tdsIsolation isoLevel`
where `type isoLevel uint8`

File: mssql_go18.go [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+			tdsIsolation = isolationSnapshot
+		case sql.LevelSerializable:
+			tdsIsolation = isolationSerializable
+		case sql.LevelLinearizable:
+			return nil, errors.New("LevelLinearizable isolation level is not supported")
+		default:
+			return nil, errors.New("Isolation level is not supported or unknown")
+		}
+	}
+	return c.begin(uint8(tdsIsolation))
```
Comment (kardianos) [inline mssql_go18.go]: Then one fewer cast here.

File: tran.go (hunk 1 above)
Comment (denisenkom) [inline tran.go]: sounds good, will do

**Pull request merged**